auth-service/src/main/kotlin/uk/gov/communities/delta/auth/saml/SAMLTokenService.kt

```diff
@@ -0,0 +1,178 @@
+package uk.gov.communities.delta.auth.saml
+
+import org.opensaml.core.config.InitializationService
+import org.opensaml.core.xml.config.XMLObjectProviderRegistrySupport
+import org.opensaml.core.xml.schema.XSString
+import org.opensaml.core.xml.schema.impl.XSStringBuilder
+import org.opensaml.saml.common.SAMLVersion
+import org.opensaml.saml.common.xml.SAMLConstants
+import org.opensaml.saml.saml2.core.*
+import org.opensaml.saml.saml2.core.impl.*
+import org.opensaml.security.x509.BasicX509Credential
+import org.slf4j.LoggerFactory
+import uk.gov.communities.delta.auth.security.DeltaLdapPrincipal
+import java.io.StringWriter
+import java.time.Instant
+import java.util.*
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.xml.namespace.QName
+import javax.xml.transform.OutputKeys
+import javax.xml.transform.TransformerFactory
+import javax.xml.transform.dom.DOMSource
+import javax.xml.transform.stream.StreamResult
+
+
+class SAMLTokenService(basicCredentials: BasicX509Credential) {
+    private val samlAssertionSigner: SAMLAssertionSigner
+    private val logger = LoggerFactory.getLogger(this.javaClass)
+
+    init {
+        samlAssertionSigner = SAMLAssertionSigner(basicCredentials)
+    }
+
+    fun generate(user: DeltaLdapPrincipal, validFrom: Instant, validTo: Instant): String {
```

Comment:
Can we usefully unit test this?

Reply:
There's a test that checks the controller endpoint with fake LDAP security and makes sure a token is generated, but the generation code itself isn't meaningfully tested. We could add unit tests, but I don't think it's that valuable, whether it works with MarkLogic is the only thing that really matters and we can only do that in end to end testing.